packages/cozy-client/src/CozyClient.js

```diff
@@ -1492,7 +1493,10 @@ instantiation of the client.`
     }
 
     this.instanceOptions = root.dataset.cozy
-      ? JSON.parse(root.dataset.cozy)
+      ? {
+          ...JSON.parse(root.dataset.cozy),
+          ...omit({ ...root.dataset }, 'cozy')
+        }
```

Comment:
Would something like this work?
```js
const {cozy, ...dataset} = root.dataset;

this.instanceOptions = {...cozy, ...dataset}
```
if so it would allow to delete the ternary operator and the usage of omit()

Reply:
I think this one should works better: 
```jsx
const {cozy = '{}', ...dataset} = root.dataset;
this.instanceOptions = {...JSON.parse(cozy), ...dataset}
```